packages/react-performance/README.md

```diff
@@ -50,36 +86,131 @@ export function LastNavigationDetails() {
 
-> This section is a work in progress. It will be filled out as helper libraries for marshalling data on the server are added to Quilt.
+```tsx
+// App.tsx
+
+import React from 'react';
+import {
+  PerformanceMark,
+  usePerformanceReport,
+} from '@shopify/react-performance';
+
+/**
+ *  The top level component of our application, it should handle rendering all top level providers and determining what pages to render
+ */
+function App() {
+  // send a performance report to `/performance_report`
+  usePerformanceReport('/performance_report');
+
+  return (
+    <>
+      {/* render our LastNavigationDetails component to give devs some useful stats */}
+      <LastNavigationDetails />
+      <Cats />
+      {/* rest of the app */}
+    </>
+  );
+}
+```
+
+Assuming our `<Cats />` component was unchanged from the example above, our code would send a performance report to the `/performance_report` url as JSON. Generally, the easiest way to set up a server to make use of the reported data is to use one of our server-side companion libraries.
+
+### Parsing report JSON server-side
+
+#### Using Node (`@shopify/koa-performance`)
+
+For Node services based on `Koa`, we provide [@shopify/koa-performance](https://www.npmjs.com/package/@shopify/koa-performance) to parse `PerformanceReport` payloads and send them forward to a [StatsD](https://github.com/statsd/statsd) endpoint as `distribution`s.
+
+#### Using Rails (`quilt_rails`)
+
+For Rails services we provide [quilt_rails](https://www.npmjs.com/package/@shopify/koa-performance)'s `Quilt::Performance::Reportable` mixin for parsing `PerformanceReport` payloads and sending them forward to a [StatsD](https://github.com/statsd/statsd) endpoint as `distribution`s.
+
+### Tracking _usable_ vs _complete_
+
+In a more complex application with multiple pages and calls for remote data, it is helpful to use multiple `<PerformanceMark />` components to track data for when the page is _usable_ vs _complete_. As a rule of thumb, consider a page _usable_ when the top-level React component for it has mounted, and _complete_ when all asynchronous work (such as data fetching) required for rendering the page content has been completed.
+
+For example, we can update our `<Cats />` page component from above and change it to simulate fetching data from a remote endpoint.
+
+```tsx
+// Cats.tsx
+
+import React, {useState, useEffect} from 'react';
+import {PerformanceMark} from '@shopify/react-performance';
+
+/**
+ *  Represents an entire page of our application
+ */
+export function Cats() {
+  // simulate getting data from a remote API
+  const data = useFakeRemoteData();
+
+  return (
+    <section>
+      <h1>Cats</h1>
+      {/* mark the page as usable immediately upon rendering */}
+      <PerformanceMark stage="usable" id="cats" />
+      {data && (
+        <>
+          {/* mark the page as *complete* once we have our data */}
+          <PerformanceMark stage="complete" id="cats" />
```

Comment:
Should we be defaulting to just using the hook?

Reply:
I'm actually thinking of getting rid of the hook for this from the public API, I think you generally care where you put these and want to be able to conditionally render them, so they feel more like they want to be a component to me.

Vice versa, I don't see a reason to ever not use the hook version of `usePerformanceReport` so I'm using it in all the examples.

Reply:
If we make the hook possible to use for "unfinished" cases, wouldn't that resolve the conditional rendering?

Reply:
I'm not sure I prefer adding more API to the hook rather than just making the component the only API for it.

Reply:
In a moment I'll push a big overhaul, part of which suggests only using the hook for `usePerformanceReport` and the component for `<PerformanceMark />`, let me know how you feel about it